File: SearchAlgorithms/BinarySearch/binarySearch.c
#include <stdio.h>
int binarySearch(int arr[5],int key,int size)
{
    int first,last,mid;
    first=0;
    last=size-1;

    while (first<=last)
    {
       mid=(first+last)/2;
       if (arr[mid]==key)
       {
            return mid;
       }
       else if (key<arr[mid])
        last=mid-1;
       else
        first=mid+1;

    }
    return -1;
}

int main()
{
   int arr[5], key,size, i,result;
   printf("Enter array size: \n");
   scanf("%d",&size);

   printf("Enter array values\n");
   for (i=0;i<size;i++)
   {
       scanf("%d",&arr[i]);
   }

   printf("Enter search key: \n");
   scanf("%d",&key);

   result=binarySearch(arr,key,size);

   printf("Search key index = %d",result);

   return 0;
    }

File: Search Algorithms/BinarySearch.java
import java.util.Scanner;

class BinarySearch { 

    int binarySearch(int arr[], int l, int r, int value) 
    { 
        if (r >= l) { 
            int mid = l + (r - l) / 2; 
            if (arr[mid] == value) 
                return mid; 
            if (arr[mid] > value) 
                return binarySearch(arr, l, mid - 1, value); 
            return binarySearch(arr, mid + 1, r, value); 
        } 
        return -1; 
    } 

    public static void main(String args[]) 
    { 
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the number you want to search:");
        int value = sc.nextInt();

        BinarySearch binarySearch = new BinarySearch(); 
        int arr[] = { 2, 3, 4, 10, 40 }; 
        int n = arr.length; 
        int result = binarySearch.binarySearch(arr, 0, n - 1, value); 
        if (result == -1) 
            System.out.println("Not Found"); 
        else
            System.out.println("Found at index :" + result); 
    } 
} 
